App/App.xaml.cs

```diff
@@ -24,22 +27,51 @@ public partial class App : Application
     private readonly IServiceProvider _services;
 
     private bool _handleWindowClosed = true;
+    private const string MutagenControllerConfigSection = "MutagenController";
+
+    private const string logTemplate =
+        "{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz} [{Level:u3}] {SourceContext} - {Message:lj}{NewLine}{Exception}";
 
 #if !DEBUG
-    private const string MutagenControllerConfigSection = "AppMutagenController";
+    private const string ConfigSubKey = @"SOFTWARE\Coder Desktop\App";
+    private const string logFilename = "app.log";
 #else
-    private const string MutagenControllerConfigSection = "DebugAppMutagenController";
+    private const string ConfigSubKey = @"SOFTWARE\Coder Desktop\DebugApp";
+    private const string logFilename = "debug-app.log";
 #endif
 
+    private readonly ILogger<App> _logger;
+
     public App()
     {
         var builder = Host.CreateApplicationBuilder();
 
         (builder.Configuration as IConfigurationBuilder).Add(
-            new RegistryConfigurationSource(Registry.LocalMachine, @"SOFTWARE\Coder Desktop"));
+            new RegistryConfigurationSource(Registry.LocalMachine, ConfigSubKey));
 
         var services = builder.Services;
 
+        // Logging
+        builder.Services.AddSerilog((_, loggerConfig) =>
+        {
+            loggerConfig.ReadFrom.Configuration(builder.Configuration);
+            var sinkConfig = builder.Configuration.GetSection("Serilog").GetSection("WriteTo");
+            if (!sinkConfig.GetChildren().Any())
+            {
+                // no log sink defined in the registry, so we'll add one here.
+                // We can't generally define these in the registry because we don't
```

Comment:
We should add a new registry key under HKEY_CURRENT_USER instead of adding it to the machine-wide config

Reply:
I'm not sure that helps---the registry keys get written by the installer, and I don't think we can count on the fact that the user that runs the installer will be the user that runs Coder Desktop.

Reply:
I'm suggesting that the installer shouldn't write anything to any user-specific keys, and the app should just default to these settings instead when there's nothing configured